packages/eth-json-rpc-middleware/test/util/helpers.ts

```diff
@@ -1,9 +1,31 @@
-import type { SafeEventEmitterProvider } from '@metamask/eth-json-rpc-provider';
-import type { JsonRpcMiddleware } from '@metamask/json-rpc-engine';
+import { PollingBlockTracker } from '@metamask/eth-block-tracker';
+import {
+  providerFromEngine,
+  type SafeEventEmitterProvider,
+} from '@metamask/eth-json-rpc-provider';
+import {
+  JsonRpcEngine,
+  type JsonRpcMiddleware,
+} from '@metamask/json-rpc-engine';
 import type { Json, JsonRpcParams, JsonRpcRequest } from '@metamask/utils';
 import { klona } from 'klona/full';
 import { isDeepStrictEqual } from 'util';
 
+export const createRequest = <
+  Input extends Partial<JsonRpcRequest>,
+  Output extends Input & JsonRpcRequest,
+>(
+  request: Input,
+): Output => {
+  return {
+    jsonrpc: '2.0',
+    id: request.id ?? '1',
+    method: request.method ?? 'test_request',
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    params: request.params === undefined ? [] : request.params,
+  } as Output;
```

Comment:
Why are we forcing `params` to be present and why are we forcing the return type to be `Output`? Is this here to clean up some things but it would take a larger refactor to truly fix this?

Reply:
Re: `params`, `eth_*` JSON-RPC method parameters are conventionally always arrays, and an empty array is equivalent to no params. It's more succinct to default to something than to omit the field, so here we are.

Re: casting the return value to be `Output`, we need the Output to be a `JsonRpcRequest` and that's what we're doing here. That said, I did make the params `Json[]`, which is more accurate.